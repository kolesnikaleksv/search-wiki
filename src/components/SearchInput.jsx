import { Autocomplete, Button, TextField } from '@mui/material';
import { useState } from 'react';
import useSearch from '../hooks/useSearch';
import useDebounce from '../hooks/useDebounce';
import { Link } from 'react-router-dom';

export default function SearchInput() {
  const [value, setValue] = useState(null);
  const {articles} = useSearch(useDebounce(value));

  return (
    <div className='search box' style={{display: 'flex', flexDirection: "row", alignItems: 'center'}}>
      <Autocomplete
        disablePortal
        id="combo-box-demo"
        options={articles}
        isOptionEqualToValue={(option, value) => option.value === value.value}
        sx={{ width: 300, my: "20px"}}
        onInputChange={(event, newInputValue) => {
          setValue(newInputValue);
        }}
        //                       renderOption add wiki-link to each option item
        // renderOption={(props, option) => (
        //   <li {...props}>
        //     <a href={option.url} target="_blank" rel="noopener noreferrer">
        //      { option.label ? option.label : 'Type three or more caracters' }
        //     </a>
        //   </li>
        // )}
        renderInput={(params) => <TextField {...params} label="Wikipedia" />}
      />
      <Button style={{height: "40px", marginLeft: "20px"}} variant="contained">
        <Link style={{textDecoration: 'none', color: 'inherit'}} to={`/search?query=${value}`}>See all results</Link>
      </Button>
    </div>
  );
}