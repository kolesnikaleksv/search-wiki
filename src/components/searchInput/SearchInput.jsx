import { Autocomplete, TextField } from '@mui/material';

export default function SearchInput({articles, setValue}) {
  return (
    <Autocomplete
      disablePortal
      id="combo-box-demo"
      options={articles}
      isOptionEqualToValue={(option, value) => option.value === value.value}
      sx={{ width: 300, my: "20px"}}
      onInputChange={(event, newInputValue) => {
        setValue(newInputValue);
      }}
      renderInput={(params) => <TextField {...params} label="Wikipedia" />}
    />
  );
}