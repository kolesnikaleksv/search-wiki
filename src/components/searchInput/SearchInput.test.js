import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SearchInput from './SearchInput';

describe('SearchInput component', () => {
  const articles = [
    { value: 'Article 1', label: 'Article 1' },
    { value: 'Article 2', label: 'Article 2' },
    { value: 'Article 3', label: 'Article 3' },
  ];
  let sut;
  let props;

  it('renders autocomplete component', () => {
    const { getByLabelText } = render(<SearchInput articles={articles} setValue={() => {}} />);
    const autocompleteInput = getByLabelText('Wikipedia');
    expect(autocompleteInput).toBeInTheDocument();
  });

  it('updates input value when typing', () => {
    const setValue = jest.fn();
    const { getByLabelText } = render(<SearchInput articles={articles} setValue={setValue} />);
    const autocompleteInput = getByLabelText('Wikipedia');
    fireEvent.change(autocompleteInput, { target: { value: 'search term' } });
    expect(setValue).toHaveBeenCalledWith('search term');
  });

  // Test from course

  it('renders SearchInput component', () => {
    sut = render(<SearchInput {...props} />);
    expect(sut).toMatchSnapshot();
  });
});