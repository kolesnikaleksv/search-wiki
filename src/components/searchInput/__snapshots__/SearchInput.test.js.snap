// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component renders SearchInput component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-t79x25-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="combo-box-demo"
            id="combo-box-demo-label"
          >
            Wikipedia
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="combo-box-demo"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Wikipedia
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-t79x25-MuiAutocomplete-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="combo-box-demo"
          id="combo-box-demo-label"
        >
          Wikipedia
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id="combo-box-demo"
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
          <div
            class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Wikipedia
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
